98_exercises.Rmd

```diff
@@ -291,24 +291,87 @@ See the OMA book for examples.
+   `ExperimentList` of your TreeSEs, the second is colData and the third is
+   sampleMap
+5. Make sure that the MAE experiments are identical to the original TreeSEs.
+
+Good job! Now you are aware of how MAEs are built and we can proceed to some
+explorative exercises.
 
 ### Multi-omic data exploration
 
-1. Load experimental dataset from microbiomeDataSets (e.g., HintikkaXOData).
-2. Analyze correlations between experiments. (Taxa vs lipids, Taxa vs biomarkers, Lipids vs biomarkers)
-3. Agglomerate taxa data.
-4. Apply CLR to taxa data, apply log10 to lipids and biomarkers.
-5. Perform cross-correlation analyses and visualize results with heatmaps. (Use Spearman coefficients)
-6. Is there significant correlations? Interpret your results.
-
-Useful functions: pheatmap, ComplexHeatMap::Heatmap, ggplot, transformCounts, testExperimentCrossAssociation
+Here we learn some explorative methods and apply to the HintikkaXOData. If you
+solved the previous exercise, you can use the MAE you created and skip the first
+step.
+
+1. Store the HintikkaXOData MAE into a variable named `mae`
+2. Which experiments make up the MAE? How many samples and features are contained
+   by each experiment? You can check that with `dim`, `nrow` and `ncol`
+3. Agglomerate the microbiota experiment by Genus and store it into a new TreeSE
+   named `microbiota_genus`. How many features remain after agglomerating?
+4. Add `microbiota_genus` as a new experiment to `mae`. Note that new experiments
+   can be added to a MAE through simple concatenation with `c(mae, ...)`
+   
+So far so good. You explored the MAE, agglomerated one of its experiments and
+added it to the container. Now we will perform a cross-correlation analysis
+between two of the experiments in the MAE.
+
+1. Apply a CLR transformation to the counts assay of the agglomerated microbiota
+   experiment, and a log10 transformation to both the assays of the metabolite
+   and biomarkers experiments. You can use the function `transformCounts` and
+   don't forget to specify the assay to be transformed with the argument
+   `assay.type`
+2. Analyze correlations between the agglomerated microbiota and the biomarkers
+   with `getExperimentCrossAssociation`. Don't forget to specify the experiments
+   you want to compare with the arguments `experiment1` and `experiment2`, and
+   which of their assays with `assay.type1` and `assay.type2`
+3. What does the output look like? By default, correlation is measured in terms
+   of Kendall tau coefficients. Repeat point 2, but this time change the method
+   to Spearman coefficients.
+4. Are you able to infer significance from the output? In order to also obtain
+   p-values from the cross-correlation analysis, repeat point 2 with the
+   additional argument `test_significance = TRUE`
+5. Visualize results with a heatmap similarly to the example in section
+   \@ref(cross-correlation). Do you see any significant correlations?
+   Interpret your results.
```

Comment:
Let us have separate exercises on adding transformations or other manipulations in MAE.

The cross correlation experiment could be simplified by not including new transformations. 

Let us keep each exercise more focused on learning one thing well. We can have multiple exercises that partially complement each other.

Reply:
-> so this exercise could be splitted a bit.